### PR TITLE
graphqlbackend: reduce usage of `database.Mocks`

### PR DESCRIPTION
**Pull request by unknwon**
This PR reduces usage of `database.Mocks` in the `cmd/frontend/graphqlbackend` package.

---

Part of #26113

### CODE REVIEW

File: cmd/frontend/backend/repos.go [hunk 1]
```diff
@@ -37,9 +37,13 @@ func (e ErrRepoSeeOther) Error() string {
...
 
-var Repos = &repos{
-	store: database.GlobalRepos,
-	cache: dbcache.NewIndexableReposLister(database.GlobalRepos),
+var Repos = NewRepos(database.GlobalRepos)
+
+func NewRepos(repoStore database.RepoStore) *repos {
+	return &repos{
+		store: repoStore,
+		cache: dbcache.NewIndexableReposLister(repoStore),
```
Comment (keegancsmith) [inline cmd/frontend/backend/repos.go:45]: you can't create this on the request path, we want this cache to be static across requests

Comment (unknwon) [reply cmd/frontend/backend/repos.go:45]: Gotcha... let me think about it.

Comment (camdencheek) [reply cmd/frontend/backend/repos.go:45]: Ah yes. This is the last remaining holdout of `database.GlobalRepos`. There are a lot of things that depend on it that don't have a database that can be passed in, and I've failed multiple times to change that. I like the approach of keeping th global for now, but creating a constructor that can have a db passed in.

File: cmd/frontend/backend/repos.go [hunk 2]
```diff
@@ -42,6 +42,13 @@ var Repos = &repos{
 	cache: dbcache.NewIndexableReposLister(database.GlobalRepos),
 }
 
+func NewRepos(repoStore database.RepoStore) *repos {
+	return &repos{
+		store: repoStore,
+		cache: Repos.cache,
```
Comment (unknwon) [inline cmd/frontend/backend/repos.go:48]: @keegancsmith PTAL this approach, essentially reusing the global cache for now and only accepts a new  `database.RepoStore`.

**Review by keegancsmith — APPROVED** [review]
LGTM, but I'd add documentation to NewRepos pointing out the hidden global state / ignoring of the repostore for that cache.

**Review by camdencheek — APPROVED** [review]

**Pull request merged**